src/SimplePromiseQueue.ts

```diff
@@ -1,25 +1,26 @@
 // tslint:disable: variable-name
 export default class SimplePromiseQueue {
   private readonly _queue: Array<() => Promise<void>> = []
-  private _flushing = false
+  private _runner: Promise<void> | null = null
 
   public enqueue(task: () => Promise<void>) {
     this._queue.push(task)
-    if (!this._flushing) { return this.flushQueue() }
-    return Promise.resolve()
+    this.join()
```

Comment:
did you mean to return this promise?

Reply:
No (we discussed this in person as well). We want `enqueue` to return immediately and don't want it to return a Promise which would make the enqueue operation wait for the entire queue to flush.

To make it clearer that we are starting queue execution here, I renamed `join` back to `flushQueue`.